pymodbus/datastore/simulator.py

```diff
@@ -637,14 +637,17 @@ def action_random(cls, registers, inx, cell, _kwargs):
         :meta private:
         """
         if cell.type in (CellType.BITS, CellType.UINT16):
-            registers[inx].value = random.randint(1, 65536)
+            minval,maxval = cls.validate_range(_kwargs, 1, 65536)
```

Comment:
As the values are UINT, I wanted to avoid the case when a negative value is set. I have also considered the hard-coded values in the original code.

I can move the validation to the loading point if you suggest where it should go (expect somewhere close to `json.load()`). Or I can remove them completely. Just decide.

Reply:
the values of a register is UINT16, but the configured values can be a lot more depending on their type (e.g. a int32 is stored in 2 UINT16).

If you move it, then you need to move it into the configuration check, where we validate the actions.

I would remove that validation, as it is anyhow going to change in the next version.